### ISSUE TITLE
Possible to validate font file?

### ISSUE THREAD

**Issue by caternuson** [adafruit/Adafruit_CircuitPython_framebuf#18]
There have been numerous cases of the font5x8.bin file being downloaded wrong, for ex. a wget that results in a file full of HTML. Everything runs without errors, but text gets rendered as garbled pixels.

Any way the font file could be validated?

**Comment by ladyada** [comment]
we could check the filesize?

**Comment by caternuson** [comment]
Looks promising. Guess this is the spec for the file:
```
        # is font5x8.bin).  The font format is a binary file with the following
        # format:
        # - 1 unsigned byte: font character width in pixels
        # - 1 unsigned byte: font character height in pixels
        # - x bytes: font data, in ASCII order covering all 255 characters.
        #            Each character should have a byte for each pixel column of
        #            data (i.e. a 5x8 font has 5 bytes per character).
```
so file size needs to be `2 + 256 * width` bytes.

Quick check with `font5x8.bin`. CPython returns a class:
```python
pi@pizerow:~/oled_test $ python3
Python 3.7.3 (default, Apr  3 2019, 05:39:12) 
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> info = os.stat('font5x8.bin')
>>> type(info)
<class 'os.stat_result'>
>>> info
os.stat_result(st_mode=33188, st_ino=125186, st_dev=45826, st_nlink=1, st_uid=1000, st_gid=1000, st_size=1282, st_atime=1565727116, st_mtime=1565727116, st_ctime=1565727138)>>> info.st_size
1282
>>> info[6]
1282
>>> 
```
but with CP, it's just a tuple.
```python
Adafruit CircuitPython 4.0.2 on 2019-06-27; Adafruit ItsyBitsy M4 Express with samd51g19
>>> import os
>>> info = os.stat('font5x8.bin')
>>> type(info)
<class 'tuple'>
>>> info
(32768, 0, 0, 0, 0, 0, 1282, 1565730990, 1565730990, 1565730990)
>>> info[6]
1282
>>> 
```
The indices line up, so hopefully `os.stat('font5x8.bin')[6]` would be OK to use for both Blinka and CP support.

**Comment by ladyada** [comment]
yep! lets add a PR,can throw an exception if the font file doesnt match :)

**Comment by caternuson** [comment]
want me to bump / release? then i guess this can be closed and we'll see what happens once this gets out in the wild.

**Comment by caternuson** [comment]
Fixed with #19
